### DISCUSSION TITLE
wire:ignore misbehaving (?) on $wire call from recursively rendered alpine components.

### DISCUSSION THREAD

**Discussion by waterbuckit** [livewire/livewire#4217] [1. Help]
Hi, I've got some nested data, like so, that relates to a top-level location:
```php
[
     [
       "id" => 682,
       "name" => "New Sub",
       "location_id" => 1602,
       "sub_location_id" => null,
       "children" => [
         [
           "id" => 683,
           "name" => "Nested sub",
           "location_id" => 1602,
           "sub_location_id" => 682,
           "children" => [
             [
               "id" => 684,
               "name" => "Nested sub",
               "location_id" => 1602,
               "sub_location_id" => 683,
               "children" => [],
             ],
           ],
         ],
       ],
     ],
   ]
```
This can be arbitrarily deep, so I'm handling the rendering on the clientside using Alpine. I handle this (in a stripped down fashion) like so, my top level livewire component (full page) (you can tell that from the wire:key, I only want this to be re-rendered if the top level selected location changes): 
```html
<div>
   @if($selectedLocation->getKey())
        <div wire:key="selected-location-{{$selectedLocation->id}}">
            <div>
                <x-cleaning::location-hierarchy :location="$selectedLocation"/> 
             </div>
         </div>
   @else
       <div>Blah</div>
   @endif
</div>
```
My `location-hierarchy` component contains the alpine component, which has a wire:ignore so that it's left alone if the key does not change - there's an html getter for grabbing the recursively rendered content:
```html
@props(['location'])
<div
    wire:ignore
    x-data="
    {
        'selectedLocationId' : @entangle('selectedLocationId'),
        'selectedLocation' : {{$location->toJson()}},
        'selectedSubLocationId' : @entangle('selectedSubLocationId'),
        'locationTree' : null,
        'loadLocation' : async function(){
            const locationTreeNew = await $wire.loadLocation(this.selectedLocation.id, false, true); // returns back above data structure
            this.locationTree = locationTreeNew;
            this.loading = false;
        }, 
        get html(){
            return `{{Illuminate\Support\Facades\View::make('cleaning::components.sub-location-content')->render()}}`;
        }
    }
    " 
    x-on:reload-location.window="loadLocation" 
    x-init="loadLocation">
    <div>
        <a x-on:click.stop="$wire.loadLocation(selectedLocationId, false, true)">
            <span class="absolute inset-0"></span>
                <p x-text="selectedLocation.name">
                </p>
        </a>
     </div>
     <template x-if="locationTree">
        <ul>
            <template x-for="subLocationTreeNode in locationTree" :key="subLocationTreeNode.id">
                <x-cleaning::sub-location-hierarchy />
            </template>
        </ul>
    </template>
</div>
```
The `sub-location-hierarchy` component then uses the html() getter to render its contents recursively, renders on next tick to ensure treedata is there:
```html
<li 
    x-html="
    render ? html : ''
    " 
    x-data="{ 
            open : false,
            render : false,
    }" 
    x-init="$nextTick(() => render = true)"    
    x-on:click.stop="open = !open; $wire.selectSubLocation(subLocationTreeNode.id); ">
</li>
```
Then `sub-location-content` component is rendered, which has the actual recursive part in it:
```html
<div>
    <div x-text="subLocationTreeNode.name" class="text-sm font-medium text-gray-900">
    </div>
</div>
<template x-if="subLocationTreeNode.children.length">
    <ul x-show="open" x-collapse>
        <template x-for="subLocationTreeNode in subLocationTreeNode.children" :key="subLocationTreeNode.id">
            <x-cleaning::sub-location-hierarchy />
        </template>
    </ul>
</template>
```
My issue is that clicking on the `sub-location-hierarchy` elements, runs the `$wire.selectSubLocation` call, but for some reason triggers a re-render which causes my tree to misbehave and close - at least, I'm fairly sure that this is the cause of it, as taking away the `$wire` call makes everything work fine, but I need that wire call as it updates a separate part of the page. I can't think of anything else? Anyone else have any ideas? Should I give up trying to do this with alpine? I can't render this with livewire though, since it can go many levels deep which will kill the server. Note, the `selectSubLocation` function does NOT change the selectedLocation id, therefore, the contents of that element should not change, no? 

Thanks for any help.

**Comment by joshhanley** [comment]
@waterbuckit hmm it's hard to say without seeing it running.

Any chance you could create a small example repo which reproduces the issue with as little code as possible so we can have a look?

Thanks!
